hoodie-bench/src/main/java/com/uber/hoodie/integrationsuite/sink/writer/SinkWriter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2019 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.uber.hoodie.integrationsuite.sink.writer;
+
+import java.io.IOException;
+
+/**
+ * Implementations of {@link SinkWriter} will be able to write data to the configured Sink
+ *
+ * @param <I> Data type to be written to the sink
+ */
+public interface SinkWriter<I> {
```

Comment:
I am not sure if we should invest into a hierarchy of classes for this. Can't we use Spark datasources out of box to do this.. There are ones that write avro files into dfs already, same with writing to kafka..